src/qibo/parallel.py

```diff
@@ -35,27 +39,84 @@ def parallel_execution(circuit, states, processes=None, backend=None):
         Circuit evaluation for input states.
     """
     if backend is None:  # pragma: no cover
-        from qibo.backends import GlobalBackend
-
         backend = GlobalBackend()
 
     if states is None or not isinstance(states, list):  # pragma: no cover
-        from qibo.config import raise_error
-
-        raise_error(RuntimeError, "states must be a list.")
+        raise_error(TypeError, "states must be a list.")
 
     def operation(state, circuit):
         return backend.execute_circuit(circuit, state)
 
-    from joblib import Parallel, delayed
-
     results = Parallel(n_jobs=processes, prefer="threads")(
         delayed(operation)(state, circuit) for state in states
     )
 
     return results
 
 
+def parallel_circuits_execution(
+    circuits, states=None, nshots=1000, processes=None, backend=None
+):
+    """Execute multiple circuits
+
+    Example:
+        .. code-block:: python
+
+            import qibo
+            qibo.set_backend('qibojit')
+            from qibo import models, set_threads
+            from qibo.parallel import parallel_circuits_execution
+            import numpy as np
+            # create different circuits
+            circuits = [models.QFT(n) for n in range(5, 16)]
+            # set threads to 1 per process (optional, requires tuning)
+            set_threads(1)
+            # execute in parallel
+            results = parallel_circuits_execution(circuits, processes=2)
+
+    Args:
+        circuits (list): list of circuits to execute.
+        states (optional, list): list of states to use as initial for each circuit.
+            Must have the same length as ``circuits``.
+            If one state is given it is used on all circuits.
+            If not given the default initial state on all circuits.
+        nshots (int): Number of shots when performing measurements, same for all circuits.
+        processes (int): number of processes for parallel evaluation.
```

Comment:
Please explain what `processes=None` does and which preferred mode is used threads vs ...